server.js

```diff
@@ -1,29 +1,77 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
+import express from 'express'; 
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import mongoose from 'mongoose';
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+// Defines the port the app will run on. 
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+
+
+// Thought model
+ const Thought = mongoose.model('Thought', {
+   message: { 
+     type: String,
+     required: true,
+     minLength: 5,
+     maxLength: 140
+   },
+
+   hearts: { 
+     type: Number,
+     default: 0
+   },
 
-// Start defining your routes here
+   createdAt: { 
+     type: Date, 
+     default: Date.now
+   }
+ });
+
+// ROUTES
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.send('Hello and Welcome to the Happy thoughts API, where feeling blue is not allowed! :)');
+});
+
+// GET REQUESTS 
+app.get('/thoughts', async (req, res) => { 
+  const thoughts = await Thought.find().sort({ createdAt: 'desc'}).limit(20);
+  res.json(thoughts);
+});
+
+// POST REQUESTS
+app.post('/thoughts', async (req,res) => { 
+  try { 
+    const newThought = await new Thought(req.body).save();
+    res.status(200).json(newThought);
+  } catch (error) { 
+    console.log(error);
+    res.status(400).json({ success:false, error });
+  };
+});
```

Comment:
At the moment it's possible to post likes for example by adding hearts property in the JSON body. I added 85 hearts by adding that with my message which worked. To prevent this from happening I think you can change line 54 to const newThought = await new Thought( { message }.save(); and adding const { message } = req.body; between line 53 and 54. hope that works  👍   

It also looks like you can reassign createdAt by adding that info in the JSON  body, however the thought is not posted so I guess it's fine. But I think that you could prevent that by the previous code where you only accept message property from whatever the user sends in the request.

Reply:
Thank you for spotting that! Fixed and now only 1 heart can be sent! which is sad, I want more 85 hearts !! 🙈♥️